test/FunctionalTests/ConfigTests/ConfigCustomProvidersWebTests.cs

```diff
@@ -0,0 +1,75 @@
+        {
+            await RunSite(server, runtimeFlavor, architecture, applicationBaseUrl);
+        }
+
+        [ConditionalTheory]
+        [OSSkipCondition(OperatingSystems.Linux)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
+        //[InlineData(ServerType.WebListener, RuntimeFlavor.Clr, RuntimeArchitecture.x86, "http://localhost:6201")]
+        [InlineData(ServerType.WebListener, RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:6202")]
+        //[InlineData(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x86, "http://localhost:6203")]
+        [InlineData(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6204")]
+        //[InlineData(ServerType.Kestrel, RuntimeFlavor.Clr, RuntimeArchitecture.x86, "http://localhost:6205")]
+        [InlineData(ServerType.Kestrel, RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:6206")]
+        //[InlineData(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x86, "http://localhost:6207")]
+        // Already covered by all platforms:
+        //[InlineData(ServerType.WebListener, RuntimeFlavor.CoreClr, RuntimeArchitecture.x86, "http://localhost:6208")]
+        public async Task RunSite_WindowsOnly(ServerType server, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl)
+        {
+            await RunSite(server, runtimeFlavor, architecture, applicationBaseUrl);
+        }
+
+        [ConditionalTheory]
+        [OSSkipCondition(OperatingSystems.Windows)]
+        [InlineData(ServerType.Kestrel, RuntimeFlavor.Clr, RuntimeArchitecture.x64, "http://localhost:6209")]
+        [InlineData(ServerType.Kestrel, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64, "http://localhost:6210")]
+        public async Task RunSite_NonWindowsOnly(ServerType server, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl)
+        {
+            await RunSite(server, runtimeFlavor, architecture, applicationBaseUrl);
+        }
+
+        private async Task RunSite(ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl)
+        {
+            await TestServices.RunSiteTest(
+                SiteName,
+                serverType,
+                runtimeFlavor,
+                architecture,
+                applicationBaseUrl,
+                async (httpClient, logger, token) =>
+                {
+                    var response = await RetryHelper.RetryRequest(async () =>
+                    {
+                        return await httpClient.GetAsync(string.Empty);
+                    }, logger, token, retryCount: 30);
+
+                    var responseText = await response.Content.ReadAsStringAsync();
+                    string expectedText =
+"[Hardcoded] \r\n" +
```

Comment:
Could we start commenting about what precisely we're testing in tests like these? I know they don't lend themselves well to splitting up into distinct, well named tests, but it seems like we could at least add comments here enumerating the reasons we've included some of these lines to be checked.

I know that's not something we do in any similar tests but these can be a bit inscrutable, so maybe that would be a good trend to start.